dataproc_jupyter_plugin/services/airflow.py

```diff
@@ -94,14 +95,11 @@ async def delete_job(self, composer_name, dag_id, from_page):
                     api_endpoint, headers=self.create_headers()
                 ) as response:
                     self.log.info(response)
-            cmd = f"gsutil rm gs://{bucket}/dags/dag_{dag_id}.py"
-            process = subprocess.Popen(
-                cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True
-            )
-            output, _ = process.communicate()
-            if process.returncode == 0:
+            try:
+                cmd = f"gsutil rm gs://{bucket}/dags/dag_{dag_id}.py"
+                await async_run_gsutil_subcommand(cmd)
                 return 0
-            else:
+            except subprocess.CalledProcessError as error:
                 self.log.exception("Error deleting dag")
                 raise Exception(
                     f"Error getting airflow uri: {response.reason} {await response.text()}"
```

Comment:
This exception constructor is clearly wrong; it's reporting the response text when the error came from running a gsutil command.

Moreover, however, this entire nested try-except block is redundant. The inner `try` should be removed and just rely on the outer `except` block to report the error.